packages/yuudachi/src/functions/reports/updateReport.ts

```diff
@@ -0,0 +1,63 @@
+import type { Sql } from 'postgres';
+import { container } from 'tsyringe';
+import type { CreateReport } from './createReport.js';
+import { type RawReport, transformReport } from './transformReport.js';
+import { kSQL } from '../../tokens.js';
+
+export type PatchReport = Pick<
+	Partial<CreateReport>,
+	'guildId' | 'reportId' | 'reason' | 'message' | 'referenceId' | 'attachmentUrl' | 'status'
+>;
+
+export async function updateReport(report: PatchReport) {
+	const sql = container.resolve<Sql<any>>(kSQL);
+
+	if (report.status) {
+		await sql`
+			update reports
+			set status = ${report.status}
+			where guild_id = ${report.guildId}
+				and report_id = ${report.reportId!}`;
+	}
+
+	if (report.attachmentUrl) {
+		await sql`
+			update reports
+			set attachment_url = ${report.attachmentUrl}
+			where guild_id = ${report.guildId}
+				and report_id = ${report.reportId!}`;
+	}
+
+	if (report.reason) {
+		await sql`
+			update reports
+			set reason = ${report.reason}
+			where guild_id = ${report.guildId}
+				and report_id = ${report.reportId!}`;
+	}
+
+	if (report.message) {
+		await sql`
+			update reports
+			set message_id = ${report.message.id},
+				channel_id = ${report.message.channel.id}
+			where guild_id = ${report.guildId}
+				and report_id = ${report.reportId!}`;
+	}
+
+	if (report.referenceId) {
+		await sql`
+			update reports
+			set ref_id = ${report.referenceId}
+			where guild_id = ${report.guildId}
+				and report_id = ${report.reportId!}`;
+	}
+
+	const [updatedCase] = await sql<[RawReport]>`
```

Comment:
Should probably batch everything into a transaction. Also can avoid re-selecting with `returning *` in the update queries. Perhaps also possible to consolidate this into a single query with `field = ${newField ?? 'row'}` per [documentation](https://www.postgresql.org/docs/current/sql-update.html), but I'm not sure since the documentation doesn't explain how that works.

Reply:
We can, but this is the way `updateCase` works too, so maybe this could go in another PR changing everything? Idk

Reply:
I will mark this as resolved, in the future, we can open a PR to optimize this

Reply:
#1120